java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/ServerTest.java

```diff
@@ -0,0 +1,569 @@
+        assertDiags(diags, "Error:1:0-1:9");//errors
+        assertDiags(diags, "Error:1:0-1:9", "Information:0:81-0:86", "Information:0:85-0:86");//hints
+        serverThread.stop();
+    }
+
+    private List<Diagnostic> assertDiags(List<Diagnostic>[] diags, String... expected) {
+        synchronized (diags) {
+            while (diags[0] == null) {
+                try {
+                    diags.wait();
+                } catch (InterruptedException ex) {
+                    //ignore
+                }
+            }
+            Set<String> actualDiags = diags[0].stream()
+                                               .map(d -> d.getSeverity() + ":" +
+                                                         d.getRange().getStart().getLine() + ":" + d.getRange().getStart().getCharacter() + "-" +
+                                                         d.getRange().getEnd().getLine() + ":" + d.getRange().getEnd().getCharacter())
+                                               .collect(Collectors.toSet());
+            String diagsMessage = diags[0].stream()
+                                          .map(d -> d.getSeverity() + ":" +
+                                                    d.getRange().getStart().getLine() + ":" + d.getRange().getStart().getCharacter() + "-" +
+                                                    d.getRange().getEnd().getLine() + ":" + d.getRange().getEnd().getCharacter() + ": " +
+                                                    d.getMessage())
+                                               .collect(Collectors.joining("\n"));
+            assertEquals(diagsMessage, new HashSet<>(Arrays.asList(expected)), actualDiags);
+
+            List<Diagnostic> result = diags[0];
+
+            diags[0] = null;
+
+            return result;
+        }
+    }
+
+    @Test
+    public void testNavigator() throws Exception {
+        File src = new File(getWorkDir(), "Test.java");
+        src.getParentFile().mkdirs();
+        String code = "public class Test {\n" +
+                      "    private int field;\n" +
+                      "    public void method() {\n" +
+                      "    }\n" +
+                      "    class Inner {\n" +
+                      "        public void innerMethod() {\n" +
+                      "        }\n" +
+                      "    }\n" +
+                      "}\n";
+        try (Writer w = new FileWriter(src)) {
+            w.write(code);
+        }
+        FileUtil.refreshFor(getWorkDir());
+        ServerSocket srv = new ServerSocket(0);
+        new Thread(() -> {
+            try {
+                Socket server = srv.accept();
+                Server.run(server.getInputStream(), server.getOutputStream());
+            } catch (Exception ex) {
```

Comment:
Jackpot:
warning: Can be replaced with multicatch or several catch clauses catching specific exceptions